rm-main/src/ui/tabs/torrents/tasks/status.rs

```diff
@@ -0,0 +1,107 @@
+use std::sync::{Arc, Mutex};
+
+use crate::{action::Action, ui::components::Component};
+
+use ratatui::{prelude::*, widgets::Paragraph};
+use throbber_widgets_tui::ThrobberState;
+
+pub struct StatusBar {
+    task: StatusTask,
+    task_status: CurrentTaskState,
+}
+
+impl StatusBar {
+    pub const fn new(task: StatusTask, task_status: CurrentTaskState) -> Self {
+        Self { task, task_status }
+    }
+}
+
+impl Component for StatusBar {
+    fn render(&mut self, f: &mut Frame, rect: Rect) {
+        match &self.task_status {
+            CurrentTaskState::Loading(state) => {
+                let title = match self.task {
+                    StatusTask::Add => "Adding torrent...",
```

Comment:
Maybe instead of "Adding torrent..." make it format!("Adding {}...", torrent_name[0..20])? But make sure the torrent name is actually longer than 20 chars in order to not go out of bounds